core/services/configuration_generator.rb

```diff
@@ -26,6 +27,14 @@ def self.generate_json_format(box_definitions, name, product_configs, recipes_na
     JSON.pretty_generate(role)
   end
 
+  # Chek redentials for subscription manager
+  # @param rhel_credentials redentials for subscription manager
+  def self.retrieve_subscription_credentials(rhel_credentials)
```

Comment:
I see no need for this method. Just add corresponding lines to the generate_json_format.